client/app/rollback.go

```diff
@@ -2,75 +2,63 @@ package app
 
 import (
 	"context"
-	"fmt"
 	"math/big"
 
-	"cosmossdk.io/store"
-	storemetrics "cosmossdk.io/store/metrics"
-
 	cmtcmd "github.com/cometbft/cometbft/cmd/cometbft/commands"
-	dbm "github.com/cosmos/cosmos-db"
+	cmtcfg "github.com/cometbft/cometbft/config"
 	"github.com/ethereum/go-ethereum/common"
 
 	"github.com/piplabs/story/lib/errors"
-	"github.com/piplabs/story/lib/log"
 )
 
-func Rollback(ctx context.Context, cfg Config, removeBlock bool) error {
-	engineCl, err := newEngineClient(ctx, cfg)
-	if err != nil {
-		return err
+func RollbackCometAndAppState(ctx context.Context, a *App, appCfg Config, cometCfg cmtcfg.Config, rollbackHeights int64, removeBlock bool, rollbackEVM bool) (lastHeight int64, lastHash []byte, err error) {
```

Comment:
It seems that there is no need to pass `cometCfg`. We can use `cometCfg` from `appCfg.Comet` instead. And why don't we just pass `rollbackCfg` directly instead of passing each property individually?